app/forms/declaration/RepresentativeDetails.scala

```diff
@@ -70,4 +70,8 @@ object RepresentativeDetails {
     val DirectRepresentative = "2"
     val IndirectRepresentative = "3"
   }
+
+  def isDefined(representativeDetails: RepresentativeDetails): Boolean =
```

Comment:
hmm, I think you can move this method to the class. First you can avoid to have argument so you'll have

`def isDefined(): Boolean = details.isDefined && (details.get.eori.isDefined || details.get.address.isDefined)`

So you can make whole method shorter.

Or something like this, but it's up to you:

`def isDefined(): Boolean = details.map(_.eori.isDefined).getOrElse(false) || details.map(_.address.isDefined).getOrElse(false)`

Reply:
Swings and roundabouts.. not sure either way is better. Problem of a form being made up of optional elements but the fact we need ALL of them for it to be useable / send to the backend.

Reply:
I just mean that to invoke this method you must have `RepresentativeDetails`. So this method it's related with class, you must have object to invoke this, so it can be just easier without passing arguments and usage will be easier to.

Instead of using companion object `RepresentativeDetails.isDefined(someObjectToCheck)` you will have `someObjectToCheck.isDefined`